### DISCUSSION TITLE
How to resolve ：Webresource content size is too big

### DISCUSSION THREAD

**Discussion by hyzx86** [microsoft/powerplatform-build-tools#995] [Q&A] [answered]
Dears， 

I have a custom code component, that include a external  lib [amis](https://github.com/baidu/amis). 
when execute the command `pac pcf push --publisher-prefix xxx` , the error below will throw:

```
错误: Import Solution Failed: CustomControl with name SampleNamespace.AmisTest failed to import with error: Webresource content size is too big.
    Import Solution Failed: CustomControl with name SampleNamespace.AmisTest failed to import with error: Webresource content size is too big.
```

I have two questions: 
1. Does the size limit referred to here mean the size of the entire component package, or the size of each file in the package?
2. If it's just a limit on the size of a single file, can we customize the packaging process in a config file? Like `webpack` or `vite`, we can intervene in the packaging process with a js file.

![image](https://github.com/user-attachments/assets/d7d24242-c300-4ceb-9c85-807f899d0318)

**Comment by wangzhiwei-ext** [comment, marked as answer]
Please try solution described in this article
https://rajeevpentyala.com/2024/08/13/pcf-import-solution-failed-webresource-content-size-is-too-big/

**Comment by brasov2de** [comment]
pac pcf push is actually intended for debug purposes, and creates a debug bundle.
You could try to upload a production version, like described here: https://learn.microsoft.com/en-us/power-apps/developer/component-framework/import-custom-controls?WT.mc_id=BA-MVP-5004107
Not sure if your bundle would be under 5MB then, but it could help
